--- opencv3-3.2.0-orig/modules/videoio/src/cap_libv4l.cpp	2017-02-24 15:18:29.147055849 +0800
+++ opencv3-3.2.0/modules/videoio/src/cap_libv4l.cpp	2017-02-24 15:22:09.399046503 +0800
@@ -275,7 +275,7 @@
 
 
 // default and maximum number of V4L buffers, not including last, 'special' buffer
-#define MAX_V4L_BUFFERS 10
+#define MAX_V4L_BUFFERS 4
 #define DEFAULT_V4L_BUFFERS 4
 
 // if enabled, copies data from the buffer. this uses a bit more memory,
